contracts/margin/external/ERC20/ProductionERC20Long.sol

```diff
@@ -0,0 +1,136 @@
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+*/
+
+pragma solidity 0.4.24;
+pragma experimental "v0.5.0";
+
+import { SafeMath } from "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import { Ownable } from "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import { ERC20Long } from "./ERC20Long.sol";
+import { MarginCommon } from "../../impl/MarginCommon.sol";
+import { MarginHelper } from "../lib/MarginHelper.sol";
+
+
+/**
+ * @title ProductionERC20Long
+ * @author dYdX
+ *
+ * Early production version of an ERC20Long with a token cap and a trusted closer
+ */
+contract ProductionERC20Long is
+    ERC20Long,
+    Ownable
+{
+    using SafeMath for uint256;
+
+    // ============ Events ============
+
+    event TokenCapSet(
+        uint256 tokenCap
+    );
+
+    // ============ State Variables ============
+
+    address public TRUSTED_LATE_CLOSER;
+
+    uint256 public tokenCap;
+
+    // ============ Constructor ============
+
+    constructor(
+        bytes32 positionId,
+        address margin,
+        address initialTokenHolder,
+        address[] trustedRecipients,
+        address[] trustedWithdrawers,
+        uint256 cap,
+        address trustedLateCloser
+    )
+        public
+        Ownable()
+        ERC20Long(
+            positionId,
+            margin,
+            initialTokenHolder,
+            trustedRecipients,
+            trustedWithdrawers
+        )
+    {
+        TRUSTED_LATE_CLOSER = trustedLateCloser;
+        tokenCap = cap;
+    }
+
+    // ============ Owner-Only Functions ============
+
+    function setTokenCap(
+        uint256 newCap
+    )
+        external
+        onlyOwner
+    {
+        // We do not need to require that the tokenCap is >= totalSupply_ because the cap is only
+        // checked when increasing the position. It does not prevent any other functionality
+        tokenCap = newCap;
+        emit TokenCapSet(newCap);
+    }
+
+    // ============ Internal Overriding Functions ============
+
+    function getTokenAmountOnAdd(
+        uint256 principalAdded
+    )
+        internal
+        view
+        returns (uint256)
+    {
+        uint256 tokenAmount = super.getTokenAmountOnAdd(principalAdded);
+
+        require(
+            totalSupply_.add(tokenAmount) <= tokenCap,
+            "ProductionERC20Long#getTokenAmountOnAdd: Adding tokenAmount would exceed cap"
+        );
+
+        return tokenAmount;
+    }
+
+    function closeUsingTrustedRecipient(
```

Comment:
add a comment that this is overriding `ERC20Position`'s `closeUsingTrustedRecipient`. It confused me for a moment